### PR TITLE
feat: Easier manually playlist switching, add codecs to renditions

### PR DESCRIPTION
**Pull request by brandonocasey**
Right now we don't have a great way to manually switch playlists without calling `blacklistCurrentPlaylist` and hoping for the best or using the renditions api to exclude all renditions that we don't want. This change allows us to pass a playlist into `smoothQualityChange_` and `fastQualityChange_` so that we can directly change to the playlist we want. It also adds playlist codecs to the rendition API.

### CODE REVIEW

File: src/rendition-mixin.js [hunk 1]
```diff
@@ -66,6 +66,12 @@ class Representation {
 
     this.bandwidth = playlist.attributes.BANDWIDTH;
 
+    if (playlist.attributes.CODECS) {
```
Comment (gesinger) [inline src/rendition-mixin.js:69]: For DASH content, do we want to check for the CODECS of the audio playlist (if available)?

**Comment by brandonocasey** [comment]
Added new tests and fixed tests that were failing after the rebase.

File: test/rendition-mixin.test.js [hunk 2]
```diff
@@ -41,20 +49,23 @@ const makeMockPlaylist = function(options) {
 };
 
 const makeMockVhsHandler = function(playlistOptions, handlerOptions) {
-  const mcp = {
+  const mpc = {
```
Comment (gkatsev) [inline test/rendition-mixin.test.js:52]: oops, 😆

File: src/rendition-mixin.js [hunk 3]
```diff
@@ -66,6 +67,10 @@ class Representation {
 
     this.bandwidth = playlist.attributes.BANDWIDTH;
 
+    this.codecs = codecsForPlaylist(mpc.master(), playlist);
+
+    this.playlist = playlist;
```
Comment (gesinger) [inline src/rendition-mixin.js:72]: Do we want to add the whole playlist object to the representation?

File: test/rendition-mixin.test.js [hunk 4]
```diff
@@ -331,3 +342,69 @@ QUnit.test(
...
+      bandwidth: 0,
+      uri: 'media1.m3u8',
+      codecs: 'mp4a.40.5'
+    },
+    {
+      bandwidth: 0,
+      uri: 'media2.m3u8'
+    }
+  ], {
+    smoothQualityChange: true
```
Comment (gesinger) [inline test/rendition-mixin.test.js:363]: Is this added for any specific reason?

File: test/rendition-mixin.test.js [hunk 5]
```diff
@@ -331,3 +342,69 @@ QUnit.test(
...
+
+QUnit.test('codecs attribute gets codecs from master', function(assert) {
+  const vhsHandler = makeMockVhsHandler([
+    {
+      bandwidth: 0,
+      uri: 'media0.m3u8',
+      audio: 'a1'
+    }
+  ], {
+    smoothQualityChange: true
```
Comment (gesinger) [inline test/rendition-mixin.test.js:383]: Is this added for any specific reason?

File: src/rendition-mixin.js (hunk 3 above)
Comment (brandonocasey) [reply src/rendition-mixin.js:72]: It makes it easier to pass to `smoothQualityChange`/`fastQualityChange`

File: test/rendition-mixin.test.js [hunk 6]
```diff
@@ -331,3 +342,92 @@ QUnit.test(
...
+    }
+  ]);
+
+  RenditionMixin(vhsHandler);
+
+  const renditions = vhsHandler.representations();
+
+  assert.deepEqual(renditions[0].playlist, vhsHandler.playlists.master.playlists[0], 'rendition 1 has correct codec');
+  assert.deepEqual(renditions[1].playlist, vhsHandler.playlists.master.playlists[1], 'rendition 2 has correct codec');
+  assert.deepEqual(renditions[2].playlist, vhsHandler.playlists.master.playlists[2], 'rendition 3 has no codec');
```
Comment (gesinger) [inline test/rendition-mixin.test.js:370]:
```suggestion
  assert.deepEqual(renditions[0].playlist, vhsHandler.playlists.master.playlists[0], 'rendition 1 has playlist property');
  assert.deepEqual(renditions[1].playlist, vhsHandler.playlists.master.playlists[1], 'rendition 2 has playlist property');
  assert.deepEqual(renditions[2].playlist, vhsHandler.playlists.master.playlists[2], 'rendition 3 has playlist property');
```

**Pull request merged**